### PR TITLE
site: new dark theme

### PR DESCRIPTION
**Pull request by aslilac**
- Replace our old color palette with Tailwind colors
  - I did a bunch of math to map our old colors to the closest respective colors from the default Tailwind set, so they should all look good!

- Start using the new experimental theme `roles` in a couple select places

- Slim down the set of colors exposed in the theme/colors.ts
  - The idea is to quit using this module almost entirely, by exposing a much more curated set of colors in our `NewTheme` type that covers the _vast_ majority of use cases.

- Eventually, we'll get rid of theme/colors.ts entirely, and replace the few remaining uses with just directly importing theme/tailwind.ts (and maybe we could rename _that_ to colors.ts when the time comes)

- Fix the background color that Storybook uses! It's been wrong this entire time!!

- Some other minor reorganizing, add a few extra stories for stuff

### CODE REVIEW

File: site/src/components/AuthProvider/hi.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { createContext, useContext } from "react";
```
Comment (Kira-Pilot) [inline site/src/components/AuthProvider/hi.ts:1]: Was this file added by accident perhaps

File: site/src/components/Callout/Callout.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { type FC, type ReactNode } from "react";
```
Comment (Kira-Pilot) [inline site/src/components/Callout/Callout.tsx:1]: I don't see this implemented rn. Are we planning to use this in lieu of `<Alert />`?

File: site/src/components/DeploySettingsLayout/Sidebar.tsx [hunk 3]
```diff
@@ -66,6 +66,9 @@ const SidebarNavItem: FC<
   );
 };
 
+// TODO: :axe:
+// do we really have like 5 of the same file?
```
Comment (Kira-Pilot) [inline site/src/components/DeploySettingsLayout/Sidebar.tsx:70]: :D aggravating as it might be, can we move this to a new issue or internal cleanup list?

**Comment by Kira-Pilot** [comment]
Overall, this looks great! I think we have a couple of tweaks to make (like the audit log diff font color) but I know you're going through the snapshots so I won't compile a laundry-list here.

LMK when stuff is cleaned up and I'll ✅

File: site/src/components/AuthProvider/hi.ts (hunk 1 above)
Comment (aslilac) [reply site/src/components/AuthProvider/hi.ts:1]: what would make you think that? 💀

File: site/src/pages/AuditPage/AuditPaywall.tsx [hunk 4]
```diff
@@ -10,7 +10,7 @@ export const AuditPaywall: FC = () => {
   return (
     <Paywall
       message="Audit logs"
-      description="Audit Logs allows Auditors to monitor user operations in their deployment. To use this feature, you have to upgrade your account."
+      description="Audit Logs allows Auditors to monitor user operations in their deployment. To use this feature, you need an Enterprise license."
```
Comment (Kira-Pilot) [inline site/src/pages/AuditPage/AuditPaywall.tsx:13]: :)

**Review by Kira-Pilot — APPROVED** [review]
Lookin good!

**Pull request merged**